lib/ui/text.dart

```diff
@@ -1629,10 +1629,9 @@ class TextHeightBehavior {
   /// Creates a new TextHeightBehavior object from an encoded form.
   ///
   /// See [encode] for the creation of the encoded form.
-  TextHeightBehavior.fromEncoded(int encoded)
+  const TextHeightBehavior.fromEncoded(int encoded, this.leadingDistribution)
```

Comment:
Should `leadingDistribution` be taken out of the `TextHeightBehavior` object?

`leadingDistribution` is an attribute that can be applied to individual runs of text.  But the other members of `TextHeightBehavior` can only apply to the whole paragraph.

Reply:
In the framework `TextStyle.leadingnDistribution` can be null in which case it defers to `ParagraphStyle.textHeightBehavior.leadingDistribution`. You mean we should add it as a separate field to `ParagraphStyle`?

Reply:
That does make more sense for `fromEncoded` and `encode`. But that would break some tests in the framework. Would it be possible to make these 2 methods private?

Reply:
Yes - I think it would be more consistent to make `leadingDistribution` a field in `ParagraphStyle` that can optionally be overridden in the `TextStyle`

AFAIK there isn't any need for `TextHeightBehavior.encode` and `TextHeightBehavior.fromEncoded` to be public.

Most other encode/decode methods for these objects are private (like `_encodeParagraphStyle` and `_encodeTextStyle`)

Reply:
Looking at the framework `ParagraphStyle` code, moving `leadingDistribution` to `ParagraphStyle` is going to break it. Is there a process for making such changes? Or I can hide the ugliness by making these methods private at the cost of testability.